smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/directed/CodegenDirector.java

```diff
@@ -367,8 +366,9 @@ private void registerInterceptors(C context, List<I> integrations) {
         context.writerDelegator().setInterceptors(interceptors);
     }
 
-    private void generateShapesInService(C context, ServiceShape serviceShape, Set<Shape> shapes) {
+    private void generateShapesInService(C context, ServiceShape serviceShape) {
         LOGGER.fine(() -> "Generating shapes for " + directedCodegen.getClass().getName());
+        Set<Shape> shapes = new Walker(context.model()).walkShapes(serviceShape);
```

Comment:
Technically, before `context.model()` could be different than `model` because `createContext` _could_ modify the model. But it shouldn't. Using a different model object to walk v/s topological index doesn't make sense. As such could probably use `model` in both places here, but `context.model()` is more appropriate I think.